### PR TITLE
update capability view definition

### PR DESCRIPTION
**Pull request by krehermann**
This change expands the capability view to be more useful. The main utility is both expansion to all the types in the capability registry, and friendly serialization formatting
### Supports

[deployment pr
](https://github.com/smartcontractkit/chainlink-deployments/pull/143)

### CODE REVIEW

File: deployment/common/view/v1_0/capreg_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v := CapRegView{
+				ContractMetaData: tt.fields.ContractMetaData,
+				Capabilities:     tt.fields.Capabilities,
+				Nodes:            tt.fields.Nodes,
+				Dons:             tt.fields.Dons,
+			}
+			got, err := v.DonCapabilities()
+			if (err != nil) != tt.wantErr {
```
Comment (patrickhuie19) [inline deployment/common/view/v1_0/capreg_test.go:253]: 🤨

**Review by patrickhuie19 — DISMISSED** [review]

File: deployment/common/view/v1_0/capreg.go [hunk 2]
```diff
@@ -1,45 +1,388 @@
...
+	for _, x := range node.DONIDs {
+		if x.Cmp(donId) == 0 {
+			isMember = true
+			break
+		}
+	}
+	return isMember
+}
+
+func capInDon(cap CapabilityView, don DonView) bool {
```
Comment (jmank88) [inline deployment/common/view/v1_0/capreg.go]: Would a method read easier?
```suggestion
func (don DonView) isMember(cap CapabilityView) bool {
```
Also consider that only `ID` is used:
```suggestion
func (don DonView) isMember(capID string) bool {
```
You could also make use of `slices.ContainsFunc`:
```go
func (don DonView) isMember(capID string) bool {
	return slices.ContainsFunc(don.CapabilityConfigurations, func(cfg CapabilitiesConfiguration) bool { return cfg.ID == capID })
}
```

Comment (jmank88) [reply deployment/common/view/v1_0/capreg.go]: (same for `func nodeInDon`)

Comment (krehermann) [reply deployment/common/view/v1_0/capreg.go]: done, thanks

**Review by connorwstein — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**